### DISCUSSION TITLE
How to get RTSP stream live feed to pass it to the kinesis video stream using java sdk

### DISCUSSION THREAD

**Discussion by giriDasha3** [awslabs/amazon-kinesis-video-streams-producer-sdk-java#186] [Q&A] [answered]
Hello People,

This is not a bug. 

I need assistance extracting frames from a real-time streaming protocol (RTSP) live URL and transmitting them to a Kinesis video stream using SDK for Rekognition.

I have made several attempts but have been unable to find a suitable implementation for this task.

Currently, I am using the GStreamer tool, which is functioning well. However, I would like to switch to an SDK for programmatic access.

**Comment by sirknightj** [comment, marked as answer]
Two suggestions:
- Search for an open source rtsp client, and send the frames to the java producer. We suggest creating and implementing an rtsp MediaSource.
- Use the GStreamer bindings for Java and use the rtspsrc element and the appsrc element to get the frames and send them via the java producer.

See #28 for some related discussions
